src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -767,10 +803,11 @@ private static boolean shouldCheckSeparationFromPreviousToken(DetailAST ast) {
      * @return true if it should be checked if next token is separated by whitespace,
      *      false otherwise.
      */
-    private static boolean shouldCheckSeparationFromNextToken(DetailAST ast, char nextChar) {
+    private boolean shouldCheckSeparationFromNextToken(DetailAST ast, char nextChar) {
         return !(ast.getType() == TokenTypes.LITERAL_RETURN
                     && ast.getFirstChild().getType() == TokenTypes.SEMI)
-                && ast.getType() != TokenTypes.ARRAY_INIT
+                && (ast.getType() != TokenTypes.ARRAY_INIT
+                || allowWhitespaceAroundArrayInit)
```

Comment:
@nmancus1 It gives me an error-
 `(extension) LogicConditionNeedOptimization: Condition with || at line 810 position 16 need optimization. All method calls are advised to move to end of logic expression.`
What other optimization is possible, I tried almost everything.

Reply:
This line is very important for this property to work. Otherwise in a situation like-
```
final int[] arr = new int[]{1,2,3};
```
It won't check for whitespace before `{` if property is turned off.